### PR TITLE
Add internship application stage attribute to internship class

### PR DESCRIPTION
**Pull request by rylzxc**
We want to keep track of the stage of application we are at so we know what to expect next in the application.

Let's add a new field to `AddCommand` and `EditCommand` to include `stage` attribute.

This follows the conventional way of adding and editing  an internship. As internship application is a process, allowing edits to the stage is justifiable.

Let's also edit the JUnit tests for these respective classes to reflect the changes in the main code. Respective JSON files used in testcases are updated as well.

However, not advised to follow this chunky commit method.

### CODE REVIEW

File: src/main/resources/view/InternshipListCard.fxml [hunk 1]
```diff
@@ -31,6 +31,7 @@
       <Label fx:id="role" styleClass="cell_small_label" text="\$role" />
       <Label fx:id="phone" styleClass="cell_small_label" text="\$phone" />
       <Label fx:id="email" styleClass="cell_small_label" text="\$email" />
+      <Label fx:id="stage" styleClass="cell_small_label" text="\$stage" />
```
Comment (jacobkwan) [inline src/main/resources/view/InternshipListCard.fxml:34]: personal opinion is that stage should be below role in the UI, open to discussion

File: src/test/java/seedu/workbook/storage/JsonAdaptedInternshipTest.java [hunk 2]
```diff
@@ -41,75 +44,89 @@ public void toModelType_validInternshipDetails_returnsInternship() throws Except
...
+                VALID_EMAIL, VALID_STAGE, VALID_TAGS);
         String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, Company.class.getSimpleName());
         assertThrows(IllegalValueException.class, expectedMessage, internship::toModelType);
     }
 
     @Test
     public void toModelType_invalidRole_throwsIllegalValueException() {
         JsonAdaptedInternship internship = new JsonAdaptedInternship(VALID_COMPANY, INVALID_ROLE, VALID_PHONE,
-                VALID_EMAIL, VALID_TAGS);
+                VALID_STAGE, VALID_EMAIL, VALID_TAGS);
```
Comment (jacobkwan) [inline src/test/java/seedu/workbook/storage/JsonAdaptedInternshipTest.java:63]: order of arguments should be `VALID_EMAIL` then `VALID_STAGE`? based on the `JsonAdaptedInternship` constructor. the test case will still pass due to the way it is set up but would be good to fix :)

File: src/test/java/seedu/workbook/storage/JsonAdaptedInternshipTest.java [hunk 3]
```diff
@@ -41,75 +44,89 @@ public void toModelType_validInternshipDetails_returnsInternship() throws Except
...
+                VALID_STAGE, VALID_TAGS);
         String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, Role.class.getSimpleName());
         assertThrows(IllegalValueException.class, expectedMessage, internship::toModelType);
     }
 
     @Test
     public void toModelType_invalidPhone_throwsIllegalValueException() {
         JsonAdaptedInternship internship = new JsonAdaptedInternship(VALID_COMPANY, VALID_ROLE, INVALID_PHONE,
-                VALID_EMAIL, VALID_TAGS);
+                VALID_STAGE, VALID_EMAIL, VALID_TAGS);
```
Comment (jacobkwan) [inline src/test/java/seedu/workbook/storage/JsonAdaptedInternshipTest.java:79]: same thing here

**Review by jacobkwan — CHANGES_REQUESTED** [review]
LGTM, just some minor nitpicks,

Apart from the comments left, I think you missed out two `.json` files in `JsonWorkBookStorageTest`.

The reason why the test cases using those json files are still passing is because the SUT eventually calls `JsonWorkBookStorage#readWorkBook` and asserts that an exception is thrown. 
`readWorkBook` will then throw an exception because of the missing compulsory `stage` attribute.
So we're actually detecting that instead of whatever the test was originally meant to detect (i think it was invalid fields?)
Let me know if this doesn't make sense or I am mistaken.

Perhaps could make the testing more robust? Like differentiating between missing fields and invalid fields. @glemenneo fyi Mr Testing IC

File: src/test/java/seedu/workbook/storage/JsonAdaptedInternshipTest.java (hunk 3 above)
Comment (rylzxc) [reply src/test/java/seedu/workbook/storage/JsonAdaptedInternshipTest.java]: Resolved.

File: src/test/java/seedu/workbook/storage/JsonAdaptedInternshipTest.java (hunk 2 above)
Comment (rylzxc) [reply src/test/java/seedu/workbook/storage/JsonAdaptedInternshipTest.java]: Resolved. Thank you for picking that up!

File: src/main/resources/view/InternshipListCard.fxml (hunk 1 above)
Comment (rylzxc) [reply src/main/resources/view/InternshipListCard.fxml:34]: Let's discuss this as a team.

**Comment by rylzxc** [comment]
I've fixed the ordering of parameters and have included the `stage` attribute to the missed json files.

I agree that the testing have to be more robust (for other classes similarly) , let's work on it in the coming week.

**Review by jacobkwan — APPROVED** [review]
LGTM

**Pull request merged**